### ISSUE TITLE
Form errors are not returned when using the form helper with the only property.

### ISSUE THREAD

**Issue by joowdx** [inertiajs/inertia#1639]
### Version:

- `@inertiajs/vue3` version: 1.0.9

### Describe the problem:

I needed to implement lazy loading for a collection to mitigate the slowdown during the initial page visit. However, when using the form helper for any form action, all the lazily loaded data completely disappears. To address this issue, I used the `only` property of the form helper, to make sure it gets loaded after. The caveat is that any validation errors will always returns an empty array regardless if it has any validation error or not.

### Steps to reproduce:

sample setup
#### Employee Controller
```php
public function index(Request $request)
{
    return inertia('Home/Index', [
        'employees' => Inertia::lazy(fn () => Employee::search($request->search)->paginate()),
        // rest of the data
    ]);
}

public function store(Request $request)
{
    $validated = $request->validate([
        'name' => 'required|string',
        'status' => 'required|string',
        // rest of the attributes
    ]);

    Employee::create($validated);

    return redirect()->back();
}
```

#### Home/Index.vue

```javascript
const form = useForm({
    name: '',
    status: '',
    // ...
})

const submit = () => {
    form.post(route('employees.store'), {
        preserveScroll: true,
        preserveState: true,
        only: [
            'employees',
            // all that needs to be loaded
        ]
    })
}
```

When submitting the form, it returns as successful because there will be no errors received even if you intentionally leave it null.

**Comment by joowdx** [comment]
> A workaround for this is to use `only: ['employees', 'errors']` so the errors are loaded as well.

Will definitely try this workaround later.
At first I was very confused as to what caused it to pass the validation but kind of a not really (?) It actually took me months to finally identify the reason.

**Comment by joowdx** [comment]
@RobertBoes the workaround really work wonders. However, I think the errors should be included by default rather than having to specify it explicitly in the `only` property.

**Comment by RobertBoes** [comment]
@joowdx Yeah, I agree with you. Because the errors are shared as normal props (https://github.com/inertiajs/inertia-laravel/blob/master/src/Middleware.php#L55-L57) you'd have to treat them as such while Inertia itself relies quite heavily on them internally, but it seems like a deliberate architectural choice to have it this way

**Comment by joowdx** [comment]
> Yeah interesting — maybe we need to just always include `errors` regardless of the `only` option 🤔

Yes, please. As it will make the form behave as intended when the `only` property is set.

**Comment by craigrileyuk** [comment]
You could have an `always` setting for global share on the backend.

We tend to use `flash` as a globalShare on many projects to send down flash messages and I often find myself having to `only: ["someProp","errors","flash"]`

**Comment by joowdx** [comment]
@reinink 

> Lazy data evaluation
```php
return Inertia::render('Users/Index', [
    // ALWAYS included on first visit...
    // OPTIONALLY included on partial reloads...
    // ALWAYS evaluated...
    'users' => User::get(),

    // ALWAYS included on first visit...
    // OPTIONALLY included on partial reloads...
    // ONLY evaluated when needed...
    'users' => fn () => User::get(),

    // NEVER included on first visit...
    // OPTIONALLY included on partial reloads...
    // ONLY evaluated when needed...
    'users' => Inertia::lazy(fn () => User::get()),
]);
```

Or how about the opposite of the closure method `'users' => fn () => User::get()`, which shouldn't get evaluated on the initial visit but will be on the succeeding requests? 

We have to change all the forms' `only` property to set all the variables there if we need the lazily loaded data to be evaluated every after action. Just like what @craigrileyuk said:

> I often find myself having to `only: ["someProp","errors","flash"]`

**Comment by jessarcher** [comment]
The problem with always fetching the `errors` is that sometimes you might not want that. For example, if you just wanted to reload `employees` while maintaining the existing error state.

@reinink and I had a chat about this, and he proposed this idea:

include errors in response when:
  1. there are errors
  2. or there are no errors, and the only option is `null`, or the only prop has ’errors’ included in it

**Comment by pascalbaljet** [comment]
This problem has been solved already with the introduction of `Inertia::always()` (see [this PR](https://github.com/inertiajs/inertia-laravel/pull/627)) and by changing the shared data in the Middleware so that errors are always included in the response.
